lib/openhab/core/dsl/actions.rb

```diff
@@ -10,9 +10,78 @@ module DSL
       # Module to import and steramlime access to OpenHAB actions
       #
       module Actions
-        #  def notify(user:, msg:); end
+        java_import org.openhab.core.library.types.PercentType
+        include Logging
 
-        #  def notifyAll(message); end
+        NotificationAction = nil # Avoid the (NameError) uninitialized constant error
+        OpenHAB::OSGI.services('org.openhab.core.model.script.engine.action.ActionService')&.each do |service|
+          java_import service.actionClass.to_s
+          logger.info("Loaded ACTION: #{service.actionClass}")
+        end
+        java_import org.openhab.core.model.script.actions.Exec
+        java_import org.openhab.core.model.script.actions.HTTP
+        java_import org.openhab.core.model.script.actions.Ping
+
+        #
+        # Return an OpenHAB Action object for the given scope and thing
+        #
+        # @param scope [String] The action scope
+        # @param thing_uid [String] Thing UID
+        #
+        # @return [Object] OpenHAB action
+        #
+        def action(scope, thing_uid)
```

Comment:
Are actions always attached to things?

Reply:
It just translates to $actions.get(scope, thinguid). AFAIK both are required, so yes, it seems to be always attached to things. The "scope" is which binding provides it.